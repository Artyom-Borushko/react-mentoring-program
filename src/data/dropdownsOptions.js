export const movieDropdownOptions = ['edit', 'delete'];

export const moviesSortDropdownOptions = ['release date', 'title'];
