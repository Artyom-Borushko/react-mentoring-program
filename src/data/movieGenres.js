
export const listOfGenres = ['all', 'documentary', 'comedy', 'horror', 'crime'];
