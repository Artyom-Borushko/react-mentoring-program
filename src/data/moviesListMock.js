export const listOfMoviesMock = [
    { src: '/images/movies-list/pulpFiction.png', title: "Pulp Fiction", releaseYear: "2004", relevantGenres: ['action', 'crime'], rating: '9.2', duration: '2h 22min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId1" },
    { src: '/images/movies-list/bohemianRhapsody.png', title: "Bohemian Rhapsody", releaseYear: "2003", relevantGenres: ['documentary', 'drama'], rating: '8.4', duration: '1h 27min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId2" },
    { src: '/images/movies-list/killBillVol2.png', title: "Kill Bill: Vol 2", releaseYear: "1994", relevantGenres: ['comedy', 'crime', 'oscar'], rating: '6.2', duration: '3h 12min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId3" },
    { src: '/images/movies-list/avengersWarOfInfinity.png', title: "Avengers: War of Infinity", releaseYear: "2004", relevantGenres: ['action', 'adventure'], rating: '8.2', duration: '0h 56min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId4" },
    { src: '/images/movies-list/inception.png', title: "Inception", releaseYear: "2003", relevantGenres: ['action', 'adventure', 'horror'], rating: '10.0', duration: '1h 39min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId5" },
    { src: '/images/movies-list/reservoirDogs.png', title: "Reservoir dogs", releaseYear: "1994", relevantGenres: ['crime', 'oscar'], rating: '7.7', duration: '2h 00min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId6" },
];
