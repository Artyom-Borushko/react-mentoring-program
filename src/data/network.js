export default 'http://localhost:4000/movies';
