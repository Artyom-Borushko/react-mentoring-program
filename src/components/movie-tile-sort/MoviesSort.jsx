import React from 'react';
import './moviesSort.css';
import GenreSelect from './sorters/genre-sorter/GenreSelect';
import { listOfGenres } from '../../data/movieGenres';
import SortControl from './sorters/options-sorter/SortControl';

export default function MoviesSort({
  selectedGenre, genreSelectHandler, selectedSortOption, sortControlHandler,
}) {
  return (
    <div className="movies-sort-wrapper">
      <GenreSelect
        listOfGenres={listOfGenres}
        selectedGenre={selectedGenre}
        genreSelectHandler={genreSelectHandler}
      />

      <SortControl
        selectedSortOption={selectedSortOption}
        sortControlHandler={sortControlHandler}
      />
    </div>
  );
}
