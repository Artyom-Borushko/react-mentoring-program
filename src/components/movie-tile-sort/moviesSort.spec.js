import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import MoviesSort from './MoviesSort';
import { listOfGenres } from '../../data/movieGenres';
import { moviesSortDropdownOptions } from '../../data/dropdownsOptions';

describe('MoviesSort component tests', () => {
  it('renders MoviesSort component', () => {
    const { container } = render(<MoviesSort
      selectedGenre={listOfGenres[2]}
      selectedSortOption={moviesSortDropdownOptions[1]}
    />);

    expect(container).toMatchSnapshot();
  });
});
