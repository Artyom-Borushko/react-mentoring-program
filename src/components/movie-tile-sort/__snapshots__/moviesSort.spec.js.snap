// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSort component tests renders MoviesSort component 1`] = `
<div>
  <div
    class="movies-sort-wrapper"
  >
    <div
      class="genre-selection-wrapper"
    >
      <button
        class="genre-button "
        name="all"
        type="button"
      >
        all
      </button>
      <button
        class="genre-button "
        name="documentary"
        type="button"
      >
        documentary
      </button>
      <button
        class="genre-button selected"
        name="comedy"
        type="button"
      >
        comedy
      </button>
      <button
        class="genre-button "
        name="horror"
        type="button"
      >
        horror
      </button>
      <button
        class="genre-button "
        name="crime"
        type="button"
      >
        crime
      </button>
    </div>
    <div
      class="sort-control-wrapper"
    >
      <span>
        Sort by
      </span>
      <p>
        title
      </p>
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movies-sort"
          role="button"
          src="images/icons/redTriangleDown.png"
        />
      </div>
    </div>
  </div>
</div>
`;
